### ISSUE TITLE
Expose BWSwipeCell for subclassing

### ISSUE THREAD

**Issue by emuye** [bitwit/BWSwipeRevealCell#7]
This framework is pretty useful, but BWSwipeRevealCell doesn't do quite what I'd like. I think it would be useful if you allowed BWSwipeCell to be subclassed in addition to BWSwipeRevealCell.

**Comment by bitwit** [comment]
Hi @emuye, Thanks for giving this pod a shot. What exactly do you require right now for it to be subclass-able?

I've exposed it as a public class of the module already so I'm not sure that there's anything stopping you from going ahead. It was actually my intention to always make this possible so if you are experiencing any roadblocks I would like to fix them ASAP; just let me know!

**Comment by emuye** [comment]
You're right I can technically subclass it, but the animations functions are not public (animateCellSpringRelease, animateCellSlidingDoor, animateCellSwipeThrough) so I can't override these. I'd like to be able to change the animations (add spring dampening as an example). Even if I could, releaseCompletionBlock is private so I wouldn't be able to appropriately cleanup. 

One solution might be to make the animation methods public and expose necessary animation and cleanup code via blocks. The expectation would be that subclasses should call  necessary animations and cleanup code blocks, but the subclass can create the animation however they like. As an example:

``` swift

override func animateCellSlidingDoor()
    UIView.animateWithDuration(self.animationDuration,
            delay: 0,
            usingSpringWithDamping: 0.2,
            initialSpringVelocity: 0.0,
            options: .AllowUserInteraction,
            animations: {
                      myButton.alpha = 0.0
                      super.handleCellSlidingDoor()
            },
            completion: super.handleCellSlidingDoorCompletion())
```

**Comment by bitwit** [comment]
Thanks for the food for thought! I'm working on a version 2.0.0 that further abstracts swipe handling aspects out of UITableViewCell so that it can be implemented on CollectionViewCells too (as well as any other view that needs a panel style interaction.)

This kind of feature might be appropriate for version 2.0.0 and would be consistent with a goal of providing more flexibility/customization

**Comment by bitwit** [comment]
@emuye I'm sorry I misunderstood this request while at work today, forgetting that some of these functions have defaulted to `internal`. I've prepared a commit that will expose all internal functions as public as well as the release completion block

This will be merged shortly into a version 1.0.1 and published to cocoapods
